packages/atclient/src/atclient.c

```diff
@@ -244,7 +255,7 @@ int atclient_pkam_authenticate(atclient *ctx, const char *atserver_host, const i
   /*
    * 4. Start atServer connection (kill the existing connection if it exists)
    */
-  if ((ret = atclient_start_atserver_connection(ctx, atserver_host, atserver_port)) != 0) {
+  if ((ret = atclient_start_atserver_connection(ctx, options->atserver_host, options->atserver_port)) != 0) {
```

Comment:
Before calling this function, we should check if `options->atserver_host` and `options->atserver_port` were set by the caller.

```c
if(!atclient_pkam_authenticate_options_is_atserver_host_initialized(&options) || !atclient_pkam_authenticate_options_Is_atserver_port_initialized(&options)) {
   // handle error
}
```